src/otx/algo/instance_segmentation/maskdino.py

```diff
@@ -0,0 +1,537 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Main class for OTX MaskDINO model."""
+
+from __future__ import annotations
+
+import copy
+import itertools
+from typing import Any, Callable
+
+import torch
+from torch import Tensor, nn
+from torchvision import tv_tensors
+from torchvision.models.detection.image_list import ImageList
+from torchvision.ops.roi_align import RoIAlign
+
+from otx.algo.instance_segmentation.backbones.detectron_resnet import build_resnet_backbone
+from otx.algo.instance_segmentation.heads.maskdino_head import MaskDINOHead
+from otx.algo.instance_segmentation.heads.pixel_decoder.maskdino_encoder import MaskDINOEncoder
+from otx.algo.instance_segmentation.heads.transformer_decoder.maskdino_decoder import MaskDINODecoder
+from otx.algo.instance_segmentation.losses import HungarianMatcher, SetCriterion
+from otx.algo.instance_segmentation.utils import box_ops
+from otx.algo.instance_segmentation.utils.utils import ShapeSpec
+from otx.core.data.entity.base import ImageInfo, OTXBatchLossEntity
+from otx.core.data.entity.instance_segmentation import InstanceSegBatchDataEntity, InstanceSegBatchPredEntity
+from otx.core.exporter.base import OTXModelExporter
+from otx.core.exporter.native import OTXNativeModelExporter
+from otx.core.model.instance_segmentation import ExplainableOTXInstanceSegModel
+from otx.core.utils.mask_util import polygon_to_bitmap
+
+
+class _MaskDINO(nn.Module):
+    """Main class for mask classification semantic segmentation architectures."""
+
+    def __init__(
+        self,
+        backbone: nn.Module,
+        sem_seg_head: MaskDINOHead,
+        criterion: nn.Module,
+        num_queries: int,
+        object_mask_threshold: float,
+        overlap_threshold: float,
+        size_divisibility: int,
+        test_topk_per_image: int,
+        focus_on_box: bool = False,
+        transform_eval: bool = False,
+    ):
+        super().__init__()
+        self.backbone = backbone
+        self.sem_seg_head = sem_seg_head
+        self.criterion = criterion
+        self.num_queries = num_queries
+        self.overlap_threshold = overlap_threshold
+        self.object_mask_threshold = object_mask_threshold
+        self.roi_align = RoIAlign(
+            output_size=(28, 28),
+            sampling_ratio=0,
+            aligned=True,
+            spatial_scale=1.0,
+        )
+        if size_divisibility < 0:
+            # use backbone size_divisibility if not set
+            size_divisibility = self.backbone.size_divisibility
+        self.size_divisibility = size_divisibility
+
+        # additional args
+        self.test_topk_per_image = test_topk_per_image
+
+        self.focus_on_box = focus_on_box
+        self.transform_eval = transform_eval
+
+    @classmethod
+    def from_config(cls, num_classes: int) -> _MaskDINO:
```

Comment:
Please add builders in each module, for example, `MaskDINOHead` as a builder of `MaskDINOHeadModule`, instead of hard-coding configs here.